src/components/dashboard/main-content/course-enrollments/tests/CourseEnrollments.test.jsx

```diff
@@ -1,226 +1,90 @@
-      wrapper = mount((
-        <AppContext.Provider value={{ enterpriseConfig }}>
-          <CourseEnrollments {...initialProps} />
-        </AppContext.Provider>
-      ));
-      expect(wrapper.find('.sidebar-example').exists()).toBeFalsy();
-    });
-
-    it('is shown at screen widths less than large breakpoint', () => {
-      global.innerWidth = breakpoints.small.minWidth;
-      const enterpriseConfig = {
-        uuid: 'test-enterprise-uuid',
-      };
-      wrapper = mount((
-        <AppContext.Provider value={{ enterpriseConfig }}>
-          <CourseEnrollments {...initialProps} />
-        </AppContext.Provider>
-      ));
-      expect(wrapper.find('.sidebar-example').exists()).toBeTruthy();
-    });
-  });
+jest.mock('../course-cards/mark-complete-modal/data/service');
 
-  describe('calls appropriate fetch method depending on page type', () => {
-    beforeEach(() => {
-      mockFetchCourseEnrollments.mockReset();
-      mockClearCourseEnrollments.mockReset();
-    });
-
-    it('for enterprise page', () => {
-      const uuid = 'test-enterprise-uuid';
-      const enterpriseConfig = {
-        uuid,
-      };
-      mount((
-        <AppContext.Provider value={{ enterpriseConfig }}>
-          <CourseEnrollments {...initialProps} />
-        </AppContext.Provider>
-      ));
-      expect(mockFetchCourseEnrollments.mock.calls.length).toEqual(1);
-      expect(mockFetchCourseEnrollments).toBeCalledWith({
-        uuid,
-      });
-    });
-  });
+beforeEach(() => {
+  updateCourseCompleteStatusRequest.mockImplementation(() => {});
+});
 
-  it('properly closes mark course as complete success status alert', () => {
-    const enterpriseConfig = {
-      uuid: 'test-enterprise-uuid',
-    };
-    const wrapper = mount((
+const genericMockFn = () => jest.fn();
+const store = createMockStore(configureMockStore);
+
+const enterpriseConfig = {
+  uuid: 'test-enterprise-uuid',
+};
+const completedCourseRun = createCompletedCourseRun();
+const inProgCourseRun = { ...completedCourseRun, courseRunStatus: 'in_progress' };
+
+const defaultInitialProps = defaultInitialEnrollmentProps({ genericMockFn });
+
+// TODO not sure why and if we need the sidebarComponent here
+const initProps = {
+  ...defaultInitialProps,
+  courseRuns: {
+    in_progress: [inProgCourseRun],
+    upcoming: [],
+    completed: [completedCourseRun],
+  },
+  sidebarComponent: <div className="sidebar-example" />,
+};
+
+function renderEnrollmentsComponent(initialProps) {
+  render(
+    <Provider store={store}>
       <AppContext.Provider value={{ enterpriseConfig }}>
-        <CourseEnrollments
-          {...initialProps}
-          isMarkCourseCompleteSuccess
-        />
+        <CourseEnrollments {...initialProps} />
       </AppContext.Provider>
-    ));
-    wrapper.find('.alert .btn.close').simulate('click');
-    expect(mockModifyIsMarkCourseCompleteSuccess).toBeCalledTimes(1);
+    </Provider>,
+  );
+}
+
+test('loads enrollments component', () => {
```

Comment:
really like the syntax for React Testing Library 💯

Reply:
indeed there is much to loave about it for sure, I love how Kent C. Dodds is vocal about issues enzyme has and how this tool is meant to address it. I find that many cases are not easy or possible to automated right now with this tool though since it no longer recommends using CSS classes or selectors as a way to find elements (it is possible though). Hence I think we have to go case by case and update ui elements if needed to have correct role + name combinations so we can mostly use stuff like `getbyRole('button', {name: <name>})` or any other field, which ensure that the fields used in tests are also accessible ... still much to learn and update here